akka-projection-grpc/src/main/scala/akka/persistence/EventWriter.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2009-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.LoggerOps
+import akka.actor.typed.scaladsl.adapter.TypedActorRefOps
+import akka.annotation.InternalApi
+import akka.pattern.StatusReply
+import akka.persistence.query.typed.EventEnvelope
+
+import java.util
+import java.util.UUID
+
+// FIXME move to akka-persistence-typed for access
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object EventWriter {
+
+  sealed trait Command
+  final case class Write(envelope: EventEnvelope[Any], replyTo: ActorRef[StatusReply[Done]]) extends Command
+
+  def apply(journalPluginId: String): Behavior[Command] =
+    Behaviors
+      .setup[AnyRef] { context =>
+        val writerUuid = UUID.randomUUID().toString
+        val journal = Persistence(context.system).journalFor(journalPluginId)
+        context.log.debug("Event writer for journal [{}] starting up", journalPluginId)
+
+        val waitingForResponse = new util.HashMap[(String, Long), ActorRef[StatusReply[Done]]]()
+
+        Behaviors.receiveMessage {
+          case Write(envelope, replyTo) =>
+            // FIXME trace or remove
+            context.log.debug(
+              "Writing event persistence id [{}], sequence nr [{}]",
+              envelope.persistenceId,
+              envelope.sequenceNr)
+            val repr = PersistentRepr(
+              envelope.event,
+              persistenceId = envelope.persistenceId,
```

Comment:
we might want to make it possible to rewrite the persistenceId with some kind of user defined function

Reply:
Added in [e200c1b](https://github.com/akka/akka-projection/pull/932/commits/e200c1b67ea1c4b933143f23efe466fc970a3c30)